src/SFA.DAS.ApplyService.InternalApi/Startup.cs

```diff
@@ -172,56 +175,49 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
-
-                config.For<IHttpContextAccessor>().Use<HttpContextAccessor>();
-                
-                config.For<IConfigurationService>()
-                    .Use<ConfigurationService>().Singleton()
-                    .Ctor<string>("environment").Is(_configuration["EnvironmentName"])
-                    .Ctor<string>("storageConnectionString").Is(_configuration["ConfigurationStorageConnectionString"])
-                    .Ctor<string>("version").Is(_version)
-                    .Ctor<string>("serviceName").Is(_serviceName);
-
-                config.For<IContactRepository>().Use<ContactRepository>();
-                config.For<IApplyRepository>().Use<ApplyRepository>();
-                config.For<IOrganisationRepository>().Use<OrganisationRepository>();
-                config.For<IDfeSignInService>().Use<DfeSignInService>();
-                config.For<IEmailService>().Use<EmailService.EmailService>();
-
-                // NOTE: These are SOAP Services. Their client interfaces are contained within the generated Proxy code.
-                config.For<CharityCommissionService.ISearchCharitiesV1SoapClient>().Use<CharityCommissionService.SearchCharitiesV1SoapClient>();
-                config.For<CharityCommissionApiClient>().Use<CharityCommissionApiClient>();
-                // End of SOAP Services
-
-                config.For<IKeyProvider>().Use<PlaceholderKeyProvider>();
-                config.For<IStorageService>().Use<StorageService>();
+            services.AddTransient<IHttpContextAccessor, HttpContextAccessor>();
                 
-                services.AddMediatR(typeof(CreateAccountHandler).GetTypeInfo().Assembly);
-                
-                config.Populate(services);
-            });
-
-            return container.GetInstance<IServiceProvider>();
+            services.AddSingleton<IConfigurationService>(sp => new ConfigurationService(
+                 sp.GetService<IHostingEnvironment>(),
+                 _configuration["EnvironmentName"],
+                 _configuration["ConfigurationStorageConnectionString"],
+                 _version,
+                 _serviceName));
+
+            services.AddTransient<IValidatorFactory, ValidatorFactory>();
+
+            services.AddTransient<IValidator, DateNotInFutureValidator>();
+            services.AddTransient<IValidator, DateValidator>();
+            services.AddTransient<IValidator, EmailAddressIsValidValidator>();
+            services.AddTransient<IValidator, MaxLengthValidator>();
+            services.AddTransient<IValidator, MaxWordCountValidator>();
+            services.AddTransient<IValidator, NullValidator>();
+            services.AddTransient<IValidator, RegexValidator>();
+            services.AddTransient<IValidator, RegisteredCharityNumberValidator>();
+            services.AddTransient<IValidator, RequiredValidator>();
+            services.AddTransient<IValidator, SimpleRadioNotNullValidator>();
```

Comment:
Ideally, all these IValidator wireups need replacing with some sort of "AddAllTypesThatImplement<IValidator>" call as the idea is that we can add IValidators without worrying about wiring up.  But as this will all be gone soon with the implementation of Qna Api, I'm just leaving this as a comment.